terraform/shared/shared.tf.json.template.py

```diff
@@ -166,6 +175,32 @@
                 }
             }
         },
+        **({
+               'aws_cloudformation_stack': {
+                   'chatbot': {
+                       'name': config.qualified_resource_name('chatbot'),
+                       'template_body': json.dumps({
```

Comment:
Notice how much simpler this gets without all the clutter caused by the indirections. We've essentially cut the amount of code by 90%.